src/orange/orangeinp/ConvexRegion.cc

```diff
@@ -354,6 +354,91 @@ void InfWedge::output(JsonPimpl* j) const
     to_json_pimpl(j, *this);
 }
 
+//---------------------------------------------------------------------------//
+// PARALLELEPIPED
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with a 3-vector of half-edges and three angles (in radians).
+ */
+Parallelepiped::Parallelepiped(Real3 const& half_projs,
+                               Turn alpha,
+                               Turn theta,
+                               Turn phi)
+    : hpr_{half_projs}, alpha_{alpha}, theta_{theta}, phi_{phi}
+{
+    for (auto ax : range(Axis::size_))
+    {
+        CELER_VALIDATE(hpr_[to_int(ax)] > 0,
+                       << "nonpositive half-edge - roughly along "
+                       << to_char(ax) << " axis: " << hpr_[to_int(ax)]);
+    }
+
+    CELER_VALIDATE(alpha_ >= zero_quantity() && alpha_ < Turn{0.25},
+                   << "invalid angle " << alpha_.value()
+                   << " [turns]: must be in the range [0, 0.25]");
```

Comment:
Since it can't be a full 90 degrees this is a half-open range
```suggestion
                   << " [turns]: must be in the range [0, 0.25)");
```